sw/device/lib/dif/dif_plic.c

```diff
@@ -179,13 +204,47 @@ static ptrdiff_t plic_priority_reg_offset(dif_plic_irq_id_t irq) {
   return RV_PLIC_PRIO0_REG_OFFSET + offset;
 }
 
+static void plic_reset(const dif_plic_t *plic) {
+  // Clear all of Interrupt Enable and Level/Edge registers.
+  for (int i = 0; i < plic_num_reg(); ++i) {
+    ptrdiff_t offset = RV_PLIC_IE00_REG_OFFSET + (i * sizeof(uint32_t));
+    mmio_region_write32(plic->base_addr, offset, 0);
+
+    offset = RV_PLIC_LE0_REG_OFFSET + (i * sizeof(uint32_t));
+    mmio_region_write32(plic->base_addr, offset, 0);
+  }
+
+  // Clear all of the priority registers.
+  for (int i = 0; i < plic_num_priority_reg(); ++i) {
+    ptrdiff_t offset = RV_PLIC_PRIO0_REG_OFFSET + (i * sizeof(uint32_t));
+    mmio_region_write32(plic->base_addr, offset, 0);
+  }
+
+  // Clear all of the target threshold registers.
+  for (dif_plic_target_t target = kDifPlicTargetIbex0;
+       target <= kDifPlicTargetLast; ++target) {
+    ptrdiff_t offset = plic_target_reg_offsets[target].threshold;
+    mmio_region_write32(plic->base_addr, offset, 0);
+  }
+
+  // Clear software interrupt pending register.
+  mmio_region_write32(plic->base_addr, RV_PLIC_MSIP0_REG_OFFSET, 0);
+
+  // TODO - is there any way to ensure that all CC claimed operations have been
+  // completed? If CC is non-zero and has been read (claimed), unless is writtem
+  // to with the same ID that has been claimed - it does not complete. Is there
+  // any other way to reset this register?
```

Comment:
I think it is reasonable to expect that, if the boot rom is actually handling interrupts from the PLIC, that the CC registers will only be read from in an interrupt service routine, and that ISR won't return until it has written back to the CC register. This feels like a different case to the interrupt status register in, for example, the UART (which is explicitly separate from the interrupt enable register).

Thus, I don't think we need to reset CC register manually, especially as we don't know what exactly to write back to it.

Reply:
> that the CC registers will only be read from in an interrupt service routine, and that ISR won't return until it has written back to the CC register

Well, nothing prevents it returning without servicing the interrupt (if there's a bug in the code). However, I am not sure that there is any easy way to clear the CC without knowing the claim id. But maybe this is not an issue at all, and we just assume that it has been serviced, because if its not - then the state is compromised anyway due to an earlier bug.

Reply:
>  However, I am not sure that there is any easy way to clear the CC without knowing the claim id. But maybe this is not an issue at all, and we just assume that it has been serviced, because if its not - then the state is compromised anyway due to an earlier bug.

Exactly - so can you update the comment to remove the TODO and to say we explicitly assume that CC was handled correctly and does not need resetting?

Reply:
I have removed the `TODO` and added the function description.